enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/ClientEvaluationService.java

```diff
@@ -156,24 +165,26 @@ private void processEvaluationSuccess(final IdentityVerificationEntity identityV
                         auditService.audit(document, "Document rejected because of client evaluation for user: {}", identityVerification.getUserId());
                     });
             identityVerification.setTimestampFailed(ownerId.getTimestamp());
-            saveInExistingTransaction(status ->
-                    identityVerificationService.moveToPhaseAndStatus(identityVerification, phase, IdentityVerificationStatus.REJECTED, ownerId));
+            identityVerificationService.moveToPhaseAndStatus(identityVerification, phase, IdentityVerificationStatus.REJECTED, ownerId);
         }
     }
 
-    private void processEvaluationError(final IdentityVerificationEntity identityVerification, final OwnerId ownerId, final Throwable t) {
-        logger.warn("Client evaluation failed for {} - {}", identityVerification, t.getMessage());
-        logger.debug("Client evaluation failed for {}", identityVerification, t);
+    private void processTooManyEvaluationError(final IdentityVerificationEntity identityVerification, final OwnerId ownerId) {
+        logger.warn("Client evaluation too many attempts for {} - {}", identityVerification, ownerId);
         identityVerification.setErrorDetail(IdentityVerificationEntity.ERROR_MAX_FAILED_ATTEMPTS_CLIENT_EVALUATION);
         identityVerification.setErrorOrigin(ErrorOrigin.PROCESS_LIMIT_CHECK);
         identityVerification.setTimestampFailed(ownerId.getTimestamp());
         final IdentityVerificationPhase phase = identityVerification.getPhase();
-        saveInExistingTransaction(status ->
-                identityVerificationService.moveToPhaseAndStatus(identityVerification, phase, IdentityVerificationStatus.FAILED, ownerId));
+        identityVerificationService.moveToPhaseAndStatus(identityVerification, phase, IdentityVerificationStatus.FAILED, ownerId);
     }
 
-    private void saveInExistingTransaction(final Consumer<TransactionStatus> consumer) {
-        transactionTemplate.setPropagationBehavior(TransactionDefinition.PROPAGATION_MANDATORY);
```

Comment:
Default propagation level is `REQUIRED`. I do not see any reason yet to be more strict.